ctlcmd/cmdbase.py

```diff
@@ -771,7 +772,198 @@ def globcomp(text):
+                               self.standardtitles[3]:np.float32,self.standardtitles[4]:np.float32,self.standardtitles[5]:np.float32,
+                               self.standardtitles[6]:np.float32,self.standardtitles[7]:np.float32,
+                               self.titles[1][0]:np.float64}, title=functiontype)
+      self.rootfile = file
+    elif functiontype == "timescan":
+      file.mktree("DataTree", {self.standardtitles[0]:np.float32,self.standardtitles[1]:np.int_,self.standardtitles[2]:np.float32,
+                               self.standardtitles[3]:np.float32,self.standardtitles[4]:np.float32,self.standardtitles[5]:np.float32,
+                               self.standardtitles[6]:np.float32,self.standardtitles[7]:np.float32,
+                               self.titles[2][0]:np.float64,self.titles[2][1]:np.float64,self.titles[2][2]:np.float64,
+                               self.titles[2][3]:np.float64}, title=functiontype)
+      self.rootfile = file
+    elif functiontype == "tb_levelped":
+      print("Haven't made the root tree for tb_levelped yet, will not work")
+      ##TODO: determine types of data
+    elif functiontype == "visualhscan":
+      file.mktree("DataTree", {self.standardtitles[0]:np.float32,self.standardtitles[1]:np.int_,self.standardtitles[2]:np.float32,
+                               self.standardtitles[3]:np.float32,self.standardtitles[4]:np.float32,self.standardtitles[5]:np.float32,
+                               self.standardtitles[6]:np.float32,self.standardtitles[7]:np.float322,
+                               self.titles[4][0]:np.float64,self.titles[4][1]:np.float64}, title=functiontype)
+      self.rootfile = file
+    elif functiontype == "visualzscan":
+      file.mktree("DataTree", {self.standardtitles[0]:np.float32,self.standardtitles[1]:np.int_,self.standardtitles[2]:np.float32,
+                               self.standardtitles[3]:np.float32,self.standardtitles[4]:np.float32,self.standardtitles[5]:np.float32,
+                               self.standardtitles[6]:np.float32,self.standardtitles[7]:np.float32,
+                               self.titles[5][0]:np.float64,self.titles[5][1]:np.float64,self.titles[5][2]:np.float64,
+                               self.titles[5][3]:np.float64,self.titles[5][4]:np.float64}, title=functiontype)
+      self.rootfile = file
+    else:
+      print("ERROR: Root data will not save")
+    self.saveddata = []
+    self.n=1
+    
+    timestring = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+    self.rootfile["Board_ID"]=self.board.boardid
+    self.rootfile["Board_Type"]=self.board.boardtype
+    self.rootfile["Time"]=timestring
+    ##TODO: add in lines which give the exact command used
+  
+  def fillroot(self,data,time=0.0,det_id=-100):
+    if data[0]!=-1000:
+      self.saveddata.append([time,det_id,self.gcoder.opx,self.gcoder.opy,self.gcoder.opz,
```

Comment:
We would want this to be a structured storage. To avoid structue specific "magic indices" from appearing in the code. We can for the user data to be a python **kwargs dictionary input, so that the branch name to save the various data to will be known.